parsers/go.c

```diff
@@ -678,16 +680,33 @@ static void parseStructMembers (tokenInfo *const token, const int scope)
 				}
 				readToken (token);
 			}
-			if (!isType (token, TOKEN_COMMA))
+
+			if (isType (token, TOKEN_DOT)  && (!first))
```

Comment:
I think it would be nicer to move parsing of the type outside of this loop and keep this loop only for parsing variable names. I think the way you do it now won't cause any trouble for valid input but it seems to be possible to mix variable parsing and type parsing the way it is.

Eventually it would be nice to have some "varType" (https://github.com/universal-ctags/ctags/issues/862) and report the type for all the member variables. That would mean the whole type would have to be always collected and assigned to the variable tags.